File: CodeForces/1894b/main.py
from collections import defaultdict

def inp():
    return(int(input()))
def inlt():
    return(list(map(int,input().split())))
def insr():
    return(input().strip())

def solve() :
    n = inp()
    arr = inlt()

    mp = defaultdict(int)
    ans = [1] * n
    c = 1
    for (i, x) in enumerate(arr) :
        mp[x] += 1

        if(mp[x] == 2) :
            ans[i] = min(c + 1, 3)
            c += 1

    if(c > 2) :
        print(*ans)
    else :
        print('-1')

    return            


t = inp()

for _ in range(t) :
    solve()

File: CodeForces/1894b/main.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

void solve() {
    int n;
    cin >> n;

    int a, c = 1;
    unordered_map<int, int> mp;
    vector<int> ans(n, 1);

    for(int i = 0; i < n; i++) {
        cin >> a;
        mp[a]++;
        if(mp[a] == 2) {
            ans[i] = min(c + 1, 3);
            c++;
        }
    }

    if(c > 2) {
        for(int i = 0; i < n; i++) {
            cout << ans[i] << " ";
        }
    } else {
        cout << "-1";
    }

    cout << "\n";
    
}

int main() 
{
    int t;
    cin >> t;
    while(t--) {
        solve();
    }

    return 0;
}